File: elixir/resistor-color-duo/lib/resistor_color_duo.ex
defmodule ResistorColorDuo do
  @values %{black: 0, brown: 1, red: 2, orange: 3, yellow: 4, green: 5, blue: 6, violet: 7, grey: 8, white: 9}

  @doc """
  Calculate a resistance value from two colors
  """
  @spec value(colors :: [atom]) :: integer
  def value(colors) do
    Map.get(@values, Enum.at(colors, 0)) * 10 + Map.get(@values, Enum.at(colors, 1))
  end
end

File: go/resistor-color-duo/resistor_color_duo.go
package resistorcolorduo

// Value returns the resistance value of a resistor with given colors.
func Value(colors []string) int {
	colorMap := map[string]int{"black": 0, "brown": 1, "red": 2, "orange": 3, "yellow": 4, "green": 5, "blue": 6, "violet": 7, "grey": 8, "white": 9}
    return 10 * colorMap[colors[0]] + colorMap[colors[1]]
}
